### ISSUE TITLE
Feature Request: Restart button in settings window

### ISSUE THREAD

**Issue by karpovv-boris** [ModernFlyouts-Community/ModernFlyouts#521]
When you change the app language, you see this notice:

![Screenshot_1](https://user-images.githubusercontent.com/78983523/112248303-c388b000-8c66-11eb-85c4-2885ad6adfcc.jpg)

But some of you expected this:

![111616450-f52af280-87f2-11eb-9cc5-9790b98288d2](https://user-images.githubusercontent.com/78983523/112248352-d602e980-8c66-11eb-950d-cdc82cf9068d.png)

Given that the application has a host we could tweak a little in the code to implement this for restarting the client.

**Comment by karpovv-boris** [comment]
So, I was trying to implement this, but stuck in `RegisterApplicationRestart(...)`.
Which by the way works in this context (as it shows rmlogotest.exe):

![](https://user-images.githubusercontent.com/78983523/111616990-9d40bb80-87f3-11eb-8281-2e5160303da4.png)

How can I use `RegisterApplicationRestart` using pinvoke?

**Comment by Samuel12321** [comment]
This is something I have wanted to see for a while. Thanks @karpovv-boris for your help.

**Comment by Samuel12321** [comment]
Should we move this button to beside the language change selection like powertoys?

**Comment by karpovv-boris** [comment]
Seems logical. Other settings are 'reactive'.
Right now this notice note says that there are other items in the settings that may also require you to restart the app. Which is not.
![Screenshot_1](https://user-images.githubusercontent.com/78983523/112255659-b1613e80-8c73-11eb-89bb-3900591e9a15.jpg)

**Comment by Samuel12321** [comment]
Any Updates @karpovv-boris, don't mean to bother you, just trying to plan the next few releases.

**Comment by karpovv-boris** [comment]
Still no cracking this. I'm stuck at `dllmain.LoadCLR()` in `ModernFlyoutBridge` and I don't quite understand how this is called.

**Comment by ShankarBUS** [comment]
Why do you go there? It is not necessary to look into the C++ side of the project. Everything you need is in the C# project.

**Comment by karpovv-boris** [comment]
I mean, it literally says `// But the process will exit after .NET CoreCLR shuts down anyway so... *shrug*`. Am I at wrong direction?

**Comment by ShankarBUS** [comment]
The C++ should be left untouch. It's purpose is only for loading the .NET project into a Microsoft Windows signed application.

You shouldn't modify it.

**Comment by Samuel12321** [comment]
@ShankarBUS perhaps instead of telling him where not to look, you could give him some direction on the best way to do this. 😉

**Comment by ShankarBUS** [comment]
Honestly, I am not sure. That's why I dropped the idea in first place.

I thought he might know something. I was just trying to tell him that the C++ project is not the right place.

**Comment by ShankarBUS** [comment]
Well wait a minute! Now I remember why I left this unimplemented. It is because every method that we know of start the application as unpackaged (like a normal exe), so, the app loses certain functionalities that require packaged privileges.

I tried every method I know, even the methods you two suggested. None of them will start the app as packaged.

**Comment by ShankarBUS** [comment]
We depend on certain WinRT APIs that require a package identity such APIs are app data, startup task and some others. Starting the app as unpackaged will result in serious issues such as unavailability of app data and etc.

**Comment by Samuel12321** [comment]
We may have to wait for reunion to make this possible, it seems like they are looking into solutions for this as part of their new lifecycle api's https://github.com/microsoft/ProjectReunion/issues/111

**Comment by ShankarBUS** [comment]
No, it won't help in anyway. It will solve the startup task and app activation difficulties.
In fact, we don't even need Project Reunion for this purpose. We are already doing these things ourselves and are working perfectly fine.

But for the current app data method we definitely need a package identity.

I prefer the UWP app data APIs because the data are kept safe even after app updates and stuffs. The system takes care of everything for us but at the cost of a package identity.

.NET itself have a built-in method to save and restore app data but it is not that reliable and suitable for a MSIX application.

This is a confusing situation.

**Comment by ShankarBUS** [comment]
Yeah, let's hope so
